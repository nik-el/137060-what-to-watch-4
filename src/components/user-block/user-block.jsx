import React from 'react';
import {Link} from "react-router-dom";
import PropTypes from "prop-types";
import {useSelector} from "react-redux";
import {getRestApi} from "../../reducer/data/selectors";

export const UserBlock = React.memo(function SignIn({userData}) {
  const restApi = useSelector(getRestApi);

  const signInEl =
    <div className="user-block">
      <Link className="small-movie-card__link" to={`/sign-page`}>
        Sign in
      </Link>
    </div>;

  return userData ? <Link
    to={`/my-list`}
    className="user-block"
  >
    <div className="user-block__avatar">
      <img src={restApi + userData.avatar_url} alt="User avatar" width="63" height="63"/>
    </div>
  </Link> : signInEl;
});

UserBlock.propTypes = {
  userData: PropTypes.object
};
