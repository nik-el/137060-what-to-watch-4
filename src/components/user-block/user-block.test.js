import React from 'react';
import renderer from 'react-test-renderer';

import {UserBlock} from './user-block';

const testAvatarUrl = `test`;

it(`UserBlock renders correctly`, () => {
  const tree = renderer
    .create(
        <UserBlock
          isAuth
          avatarUrl={testAvatarUrl}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

