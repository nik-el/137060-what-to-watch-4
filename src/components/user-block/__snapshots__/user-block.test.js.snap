// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock renders correctly 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="test"
      width="63"
    />
  </div>
</div>
`;
